crt/CMakeLists.txt

```
@@ -40,6 +40,12 @@ if(UNIX AND NOT APPLE)
             set(MY_ASSEMBLER_IS_TOO_OLD_FOR_512AVX ON CACHE BOOL "Disable AVX512 on old GCC that not supports it")
         endif()
 
+        string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" CMAKE_SYSTEM_PROCESSOR_LOWER)
+        if(CMAKE_SYSTEM_PROCESSOR_LOWER MATCHES "x86_64|amd64|x86|i386|i686" AND CMAKE_SIZEOF_VOID_P EQUAL 4)
```

Why doesn't AWS-LC's CMakeLists.txt handle this, vs having every consumer handle it?

---

Seems weird to include x86_64 and amd64, since those are never 32bit.
And if you take those out, it's all 32-bit architectures, so you don't need to compare pointer size
```suggestion
        if(CMAKE_SYSTEM_PROCESSOR_LOWER MATCHES "^(x86|i386|i686)$")
```

---

I think that aws-lc can still support machine without SSE2 with `-DOPENSSL_NO_ASM` to disable the assembly optimization. But, for us, we can assume we don't support 20 years old x86 CPUs.

---

https://github.com/awslabs/aws-crt-python/actions/runs/10689757684/job/29632625853

>  -- XXXXXXXXX CMAKE_SYSTEM_PROCESSOR: x86_64

https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html

>  this will correspond to the target architecture for the build, but this is not guaranteed. (E.g. on Windows, the host may be AMD64 even when using a MSVC cl compiler with a 32-bit target.)

So, I copied the logic from aws-lc. https://github.com/aws/aws-lc/blob/main/CMakeLists.txt#L791-L800
I'll copy the comments as well probably for clarity.

---

Oh yeah, I see in the [CMAKE_SYSTEM_PROCESSOR docs](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html) that it may not be correct (emphasis mine):

> In many cases, this will correspond to the target architecture for the build, **but this is not guaranteed**. (E.g. on Windows, the host may be AMD64 even when using a MSVC cl compiler with a 32-bit target.)
> 
> When cross-compiling, a CMAKE_TOOLCHAIN_FILE **should** set the CMAKE_SYSTEM_PROCESSOR variable to match target architecture that it specifies

Maybe edit the comment to be more like the docs, instead of some ramble about Docker.

> Be wary that CMAKE_SYSTEM_PROCESSOR may not correspond to the target architecture when cross-compiling. (E.g. on Windows, the host may be AMD64 even when using a MSVC cl compiler with a 32-bit target). The CMAKE_TOOLCHAIN_FILE is supposed to set it correctly, but may not have.
```

---

or just 
> CMAKE_SYSTEM_PROCESSOR is supposed to match the target architecture when cross-compiling, but this is not guaranteed. See: https://cmake.org/cmake/help/v3.30/variable/CMAKE_SYSTEM_PROCESSOR.html